MyProject/cls/Table.cs

```diff
@@ -1,59 +1,110 @@
 ﻿using System;
+using System.Windows;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Linq;
 using System.Text;
+using System.Threading;
+using static cls.Tbl;
+using System.Timers;
 
 namespace cls
 {
     public class Tbl:ISaveRecord
     {
+        public delegate void OrderChange(string change);
+       
+        public event OrderChange OrderChanged;
+
         public int TblNumber { get; private set; }
         public int Guests { get; private set; }
 
-        private int orderNumber = SetOrderNum();
-
         public ObservableCollection<string> Order = new ObservableCollection<string>();
-
-
+        
         public bool OrderExists { get; private set; }
 
-        public string Employee { get; private set; }
+        internal long _orderId { get;  set; }
 
+        private System.Timers.Timer timeOpened;
 
         public Tbl(int tableNumber)
         {
             this.TblNumber = tableNumber;
-            this.Order = null;
-        }
+            _orderId++;
 
-        internal static int SetOrderNum()
-        {
-            int lastOrderNum = 1; //get last order number from storage
-            return lastOrderNum + 1;
+            timeOpened = new System.Timers.Timer(7200000);
+            timeOpened.Interval = 7200000; // do I need to set this additionaly if the timer is already created for some time?
+            timeOpened.Elapsed += markOutstandingTable;
+            timeOpened.AutoReset = false;
+
+            OrderChanged += SaveChangesToOrder; //is this a good idea to subscribe to event when instance of a class is created?
```

Comment:
In general it is fine and used fairly a lot. The only difference is that event is used from some dependency that is injected through the ctor. e.g.:
```c#
public MyClass(IEventProvider injectedDependency)
{
    injectedDependency.Event += this.EventHandler;
}
```